### ISSUE TITLE
Wipe LoginsStorage correctly on user reset

### ISSUE THREAD

**Issue by ncalexan** [mozilla-lockbox/lockbox-android#312]
Update Android Components to a version that supports locally wiping `LoginsStorage`.  The current `LoginStorage.reset()` method does not in fact wipe local state, and the method `LoginStorage.wipe()` will result in (remote) data loss.

**This may be a breaking change version**

Related to #277 
Related to #365 

## Original Description ##

The megazord and unit testing mechanism is ready for use, but it requires Android Components 0.35.1+.  That is, #241 and #242 are blocked on upgrading.

I had a look at the upgrade path but there are non-trivial FxA changes: in particular, the multi-scope token requested at https://github.com/mozilla-lockbox/lockbox-android/blob/ee5e786149019be158676ff89036d42d1db776ca/app/src/main/java/mozilla/lockbox/store/AccountStore.kt#L121 doesn't map cleanly into the new API; on Slack, @rfk suggested that it would require N (=3) queries to produce those tokens.  That's fine -- if inefficient -- but the Lockbox code probably needs to carefully thread those finer-scoped tokens through to the end consumers, and that requires a level of knowledge of the codebase than I have.

I'm sorry to block macro changes (unit tests! megazords!) on minutiae.

**Comment by linuxwolf** [comment]
@ncalexan please provide pointers to the relevant info in the FxA client

**Comment by ncalexan** [comment]
> @ncalexan please provide pointers to the relevant info in the FxA client

Hmm, I don't really have much context here: redirecting to @vladikoff.

**Comment by linuxwolf** [comment]
*TL;DR* the `AccountStore` needs to be more selective about what scope it requests a token for, asking for `Constant.FxA.oldSyncScope` instead of the composite `Constant.FxA.scopes` when it requests the access token/encryption key.

This is necessitated by mozilla/application-services#386 (via mozilla/application-services#388) which mitigates the damage from an "over-scoped" token.

**Comment by linuxwolf** [comment]
@ncalexan I've updated the description to start with hints on what we'll have to change.  Feel free to add a comment or update the description if you're aware of other code-level changes we'll have to make.

**Comment by ncalexan** [comment]
@sashei it looks like your work in #337 addresses this completely.  I'll file a new ticket for megazording.  Can you close this one if you agree?

**Comment by sashei** [comment]
@ncalexan thanks for the heads! We did a few more upgrades than expected in #337 and I didn't realize `wipeLocal` had made it in. 

I'm going to update this ticket to reflect _using_ the `wipeLocal` method, as we've updated and done the FxA work but not switched our `LoginsStorage` interactions yet.

**Comment by devinreams** [comment]
Flagged as a P2/MVP item to do before launch since this is an important security issue/bug.
